livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py

```
@@ -754,14 +774,31 @@ async def _reconnect() -> None:
             logger.debug("reconnected to OpenAI Realtime API")
             self.emit("session_reconnected", llm.RealtimeSessionReconnectedEvent())
 
+        async def _check_session_update_done() -> None:
+            try:
+                await asyncio.wait_for(
```

Could it also be done by just adding a lock to update_instructions, update_chat_ctx, etc?

---

I think it's different, this is to make sure the update session is successful, sometimes the update failed but the session continues and user didn't get notified.

---

do we know why session.update is failing?

---

for example #3431, the update failed when reconnection due to API mismatch, causing the voice changed in the middle of conversation. It might be confusing for the users like the instruction is not applied, the voice is incorrect... we should make such error more explicit by raising an error when update failed.

---

It seems to add a lot of stuff to keep track of. So it may be harder for us to debug lol. 
Isn't just logging all the OpenAI errors enough?

---

perhaps raise an error when update option failed? users may ignore the error logs

---

btw, I think we already have the error logs from OpenAI